src/pages/Main/MainCarousel/MainCarousel.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from "react";
+import TopCarousel from "./TopCarousel/TopCarousel";
+import "./MainCarousel.scss";
+
+class MainCarousel extends Component {
+  state = {
+    topCarousel: [],
+    rightIndex: 1,
+    leftIndex: 0,
+  };
+
+  componentDidMount = () => {
+    fetch("/data/main/carousel.json", {
+      method: "GET",
+    })
+      .then(res => res.json())
+      .then(data => {
+        this.setState({ topCarousel: data.topCarousel });
+      });
+  };
+
+  // 우선 단방향 carousel 완성
+  handleRight = () => {
+    console.log(this.state.rightIndex);
+    const slideWidth = 1920;
+
+    if (this.state.rightIndex < 3) {
+      const carousels = document.querySelector(".carousels");
+      carousels.style.transform = `translateX(-${
+        this.state.rightIndex * slideWidth
+      }px)`;
+      this.setState({ ...this.state, rightIndex: this.state.rightIndex + 1 });
```

Comment:
전개구문 사용하지 않으셔도 될 것 같습니다.

```suggestion
      this.setState({ rightIndex: this.state.rightIndex + 1 });
```

Reply:
넵 반영하여 변경하였습니다:)